mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/dependencies/StrategyDependencyResolver.java

```diff
@@ -26,26 +28,46 @@
  */
 class StrategyDependencyResolver {
 
-    protected static final String SMART_TESTING_STRATEGY_PREFIX = "smart.testing.strategy.";
+    protected static final String SMART_TESTING_STRATEGY_PREFIX = Configuration.SMART_TESTING_CUSTOM_STRATEGIES + ".";
 
     private final Path propertiesPath; // TODO this could be configurable through system property and with this we need a path
+    private final String[] customStrategies;
 
     StrategyDependencyResolver(Path propertiesPath) {
         this.propertiesPath = propertiesPath;
+        this.customStrategies = null;
     }
 
     StrategyDependencyResolver() {
         this.propertiesPath = null;
+        this.customStrategies = null;
+    }
+
+    StrategyDependencyResolver(String[] customStrategies) {
+        this.propertiesPath = null;
+        this.customStrategies = customStrategies;
     }
 
     Map<String, Dependency> resolveDependencies() {
         final Properties properties = new Properties();
         properties.putAll(loadDefaultMapping());
+        properties.putAll(loadCustomStrategies());
         properties.putAll(loadFromFile());
-        properties.putAll(System.getProperties());
         return transformToDependencies(properties);
     }
 
+    private Properties loadCustomStrategies() {
+        final Properties properties = new Properties();
+
+        if (customStrategies != null) {
```

Comment:
As @dipak-pawar said - if you initialize it with an empty array instead of `null` this could be slightly nicer to digest.